lib/preview/queue.ex

```diff
@@ -56,7 +75,16 @@ defmodule Preview.Queue do
 
     case key_components(key) do
       {:ok, package, version} ->
-        create_package(package, version)
+        files = create_package(package, version)
+
+        all_versions = all_versions(package)
+        version = Version.parse!(version)
+
+        if Preview.Utils.latest_version?(version, all_versions) do
+          update_index_sitemap()
+          update_package_sitemap(package, version, files)
```

Comment:
Why do we only update the sitemaps when the latest version has been pushed?

Reply:
the index sitemap only links to package sitemaps for latest versions so I don't think there's need to update them for non-latest versions.

Reply:
Although from a second look I see we need the latest_version? for paths_for_sitemaps so do whatever is cleanest.

Reply:
I see, then we need to do the same thing when a package is deleted.

Reply:
In that case I think we should leave it as is. FWIW there's a fair amount of copy-pasta for sitemap generation (and other things too) between this and other apps, so I guess that's another reason to keep it as is for the time being.